src/txn/txn_ckpt.c

```diff
@@ -15,6 +15,153 @@ static int __checkpoint_presync(WT_SESSION_IMPL *, const char *[]);
 static int __checkpoint_tree_helper(WT_SESSION_IMPL *, const char *[]);
 static int __drop_list_execute(WT_SESSION_IMPL *session, WT_ITEM *drop_list);
 
+/*
+ * __checkpoint_flush_tier_wait --
+ *     Wait for all previous work units queued to be processed.
+ */
+static int
+__checkpoint_flush_tier_wait(WT_SESSION_IMPL *session, const char **cfg)
+{
+    WT_CONFIG_ITEM cval;
+    WT_CONNECTION_IMPL *conn;
+    uint64_t now, start, timeout;
+    int yield_count;
+
+    conn = S2C(session);
+    yield_count = 0;
+    now = start = 0;
+    /*
+     * The internal thread needs the schema lock to perform its operations and flush tier also
+     * acquires the schema lock. We cannot be waiting in this function while holding that lock or no
+     * work will get done.
+     */
+    WT_ASSERT(session, !FLD_ISSET(session->lock_flags, WT_SESSION_LOCKED_SCHEMA));
+    WT_RET(__wt_config_gets(session, cfg, "flush_tier.timeout", &cval));
+    timeout = (uint64_t)cval.val;
```

Comment:
This function is potentially called twice, once before and once after checkpointing.  My understanding is the timeout is waiting for previous operations to finish, so do we even want to look at the timeout on the call after checkpointing?

Reply:
I removed the word `previous` from the description. Both before and now, it applies to any time waiting for the flush copying to complete to the bucket.